src/server/scripts/Spells/spell_druid.cpp

```diff
@@ -459,6 +464,91 @@ class spell_dru_earthwarden : public AuraScript
     }
 };
 
+// 392124 - Embrace of the Dream
+class spell_dru_embrace_of_the_dream : public AuraScript
+{
+    bool Validate(SpellInfo const* spellInfo) override
+    {
+        return ValidateSpellInfo ({ SPELL_DRUID_EMBRACE_OF_THE_DREAM_EFFECT })
+            && ValidateSpellEffect({ { spellInfo->Id, EFFECT_2 } });
+    }
+
+    bool CheckProc(AuraEffect const* /*aurEff*/, ProcEventInfo& /*eventInfo*/)
+    {
+        return roll_chance_i(GetEffectInfo(EFFECT_2).CalcValue(GetCaster()));
+    }
+
+    void HandleProc(AuraEffect* aurEff, ProcEventInfo& /*eventInfo*/)
+    {
+        GetTarget()->CastSpell(GetTarget(), SPELL_DRUID_EMBRACE_OF_THE_DREAM_EFFECT, CastSpellExtraArgs(aurEff));
+    }
+
+    void Register() override
+    {
+        DoCheckEffectProc += AuraCheckEffectProcFn(spell_dru_embrace_of_the_dream::CheckProc, EFFECT_0, SPELL_AURA_DUMMY);
+        OnEffectProc += AuraEffectProcFn(spell_dru_embrace_of_the_dream::HandleProc, EFFECT_0, SPELL_AURA_DUMMY);
+    }
+};
+
+// 392146 - Embrace of the Dream (Chooser)
+class spell_dru_embrace_of_the_dream_chooser : public SpellScript
+{
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo
+        ({
+            SPELL_DRUID_EMBRACE_OF_THE_DREAM_HEAL,
+            SPELL_DRUID_REGROWTH,
+            SPELL_DRUID_REJUVENATION,
+            SPELL_DRUID_REJUVENATION_GERMINATION
+        });
+    }
+
+    void FilterTargets(std::list<WorldObject*>& targets)
+    {
+        targets.remove_if([&](WorldObject const* target)
+        {
+            Unit const* unitTarget = target->ToUnit();
```

Comment:
Can this EVER be false?

Reply:
if the caster is not a player, yes, some playerAI are druid class and use this exact spell, see bots in arathi brawl

Reply:
But this check doesnt check if its a player or not, its checking for unit

Reply:
spells can target gameobjects, that's why Im checking

Reply:
This spell can't target gameobjects, the TARGET_TYPE won't target GOBs.. so do you want to check for Players now or do you want to prevent targeting Gameobjects?

In both cases, your check does nothing here